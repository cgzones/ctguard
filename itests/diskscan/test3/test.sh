#!/bin/sh

set -eu

BIN=../../../src/diskscan/ctguard-diskscan
if ! [ -e ${BIN} ]; then
    echo "Could not find binary at '${BIN}'!"
    exit 1
fi

umask 0022

cleanup () {
  rm -Rf test_dir/
  rm -f test.db test.output test.output.sorted test5.txt test_diff.db
}

cleanup

chmod 640 test.conf

mkdir test_dir/
cat > test_dir/txt1.txt <<EOF
test file
this test file has many lines
1
2
3
4
5
and it contains many entries
1
2
3
4
5
6
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
the end
EOF
cat > test_dir/txt12.txt <<EOF
est file
this test file has many lines
1
2
3
4
5
and it contains many entries
1
2
3
4
5
6
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
the end
EOF
echo "test" > test_dir/txt3.txt
ln -s txt1.txt test_dir/lnk1
touch test_dir/dead
ln -s dead test_dir/lnk2
rm test_dir/dead
touch test_dir/test_rm

${BIN} -c test.conf -x -f -S

cat > test_dir/txt1.txt <<EOF
test file
this test file has many lines
9
8
7
6
5
4
3
2
1
and it contains many entries
1
2
3
4
5
6
7
8
9
text1texttexttexttexttexttexttexttexttexttexttexttexttexttext
texttext1texttexttexttexttexttexttexttexttexttexttexttexttext
texttexttext1texttexttexttexttexttexttexttexttexttexttexttext
texttexttexttext1texttexttexttexttexttexttexttexttexttexttext
texttexttexttexttext1texttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttext1texttexttexttexttexttexttexttexttext
texttexttexttexttexttexttext1texttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
the end
really
EOF
cat > test_dir/txt12.txt << EOF
test file
this test file has many lines
1
2
3
4
5
6
7
8
9
and it contains many entries
9
8
7
6
5
4
3
2
1
text1texttexttexttexttexttexttexttexttexttexttexttexttexttext
texttext1texttexttexttexttexttexttexttexttexttexttexttexttext
texttexttext1texttexttexttexttexttexttexttexttexttexttexttext
texttexttexttext1texttexttexttexttexttexttexttexttexttexttext
texttexttexttexttext1texttexttexttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
texttexttexttexttexttext1texttexttexttexttexttexttexttexttext
texttexttexttexttexttexttext1texttexttexttexttexttexttexttext
texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
the end
really
EOF
echo "test" > test_dir/txt2.txt
mkdir test_dir/testsub/
echo "test" > test_dir/testsub/txt.txt
rm test_dir/txt3.txt

cp test_dir/txt1.txt test_dir/testsub/
mv test_dir/testsub/txt1.txt test_dir/testsub/txt3.txt

mkdir test_dir/testsub/testsubsub

echo "test5" > test5.txt
mv test5.txt test_dir/
ln -s test5.txt test_dir/lnk3
rm test_dir/lnk1
rm test_dir/test_rm

${BIN} -c test.conf -x -f -S

echo "Comparing expected vs actual output:"
USERNAME=$(id -un)
GROUP=$(id -gn)
sed -e "s/ITEST_USER_REPLACEME/${USERNAME}/g" test.expected.sample | sed -e "s/ITEST_GROUP_REPLACEME/${GROUP}/g" > test.expected
LC_ALL=C sort test.output -o test.output.sorted
diff -u test.expected test.output.sorted

cleanup

echo "SUCCESS!"
