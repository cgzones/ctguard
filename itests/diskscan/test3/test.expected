 [test_dir/lnk1] : ALERT: New object
Path:            test_dir/lnk1
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 8
Mode:            lrwxrwxrwx
Target:          txt1.txt
Valid:           true
Type:            symbolic link
ALERT END


 [test_dir/lnk2] : ALERT: New object
Path:            test_dir/lnk2
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 4
Mode:            lrwxrwxrwx
Target:          dead
Valid:           false
Type:            symbolic link
ALERT END


 [test_dir/test_rm] : ALERT: New object
Path:            test_dir/test_rm
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 0
Mode:            -rw-r--r--
SHA256 checksum: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
Type:            regular file
ALERT END


 [test_dir/txt1.txt] : ALERT: New object
Path:            test_dir/txt1.txt
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 648
Mode:            -rw-r--r--
SHA256 checksum: 0af41a6baa655d68135cb7d36051f2916d9cc34d17a16cbe49aa53d9af468d09
Type:            regular file
ALERT END


 [test_dir/txt12.txt] : ALERT: New object
Path:            test_dir/txt12.txt
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 647
Mode:            -rw-r--r--
SHA256 checksum: 4b71e66a5f2139cf3fc09721c363b3e9ea81719cbf47aca0feab9dcf9e88817c
Type:            regular file
ALERT END


 [test_dir/txt3.txt] : ALERT: New object
Path:            test_dir/txt3.txt
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 5
Mode:            -rw-r--r--
SHA256 checksum: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2
Type:            regular file
ALERT END


 [test_dir/txt12.txt] : ALERT: object changed
Path:            test_dir/txt12.txt
Type:            regular file
Size changed from '647' to '676' bytes
SHA256 checksum changed from '4b71e66a5f2139cf3fc09721c363b3e9ea81719cbf47aca0feab9dcf9e88817c' to '749c0a1e0d8a0cd155fff83a27897fc26912fadf094544997ca1eb5695665e3e'
Start Diff
@@ -1,4 +1,4 @@
-est file
+test file
 this test file has many lines
 1
 2
@@ -5,21 +5,29 @@
 3
 4
 5
+6
+7
+8
+9
 and it contains many entries
+9
+8
+7
+6
+5
+4
+3
+2
 1
-2
-3
-4
-5
-6
+text1texttexttexttexttexttexttexttexttexttexttexttexttexttext
+texttext1texttexttexttexttexttexttexttexttexttexttexttexttext
+texttexttext1texttexttexttexttexttexttexttexttexttexttexttext
+texttexttexttext1texttexttexttexttexttexttexttexttexttexttext
+texttexttexttexttext1texttexttexttexttexttexttexttexttexttext
 texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
+texttexttexttexttexttext1texttexttexttexttexttexttexttexttext
+texttexttexttexttexttexttext1texttexttexttexttexttexttexttext
 texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
 the end
+really
 
End Diff
ALERT END


 [test_dir/txt1.txt] : ALERT: object changed
Path:            test_dir/txt1.txt
Type:            regular file
Size changed from '648' to '676' bytes
SHA256 checksum changed from '0af41a6baa655d68135cb7d36051f2916d9cc34d17a16cbe49aa53d9af468d09' to 'c19f94967b87bf330cc058a16a6eb18ce0bc761c5e2611bf213fc04db79a9f23'
Start Diff
@@ -1,10 +1,14 @@
 test file
 this test file has many lines
+9
+8
+7
+6
+5
+4
+3
+2
 1
-2
-3
-4
-5
 and it contains many entries
 1
 2
@@ -12,14 +16,18 @@
 4
 5
 6
+7
+8
+9
+text1texttexttexttexttexttexttexttexttexttexttexttexttexttext
+texttext1texttexttexttexttexttexttexttexttexttexttexttexttext
+texttexttext1texttexttexttexttexttexttexttexttexttexttexttext
+texttexttexttext1texttexttexttexttexttexttexttexttexttexttext
+texttexttexttexttext1texttexttexttexttexttexttexttexttexttext
 texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
+texttexttexttexttexttext1texttexttexttexttexttexttexttexttext
+texttexttexttexttexttexttext1texttexttexttexttexttexttexttext
 texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
-texttexttexttexttexttexttexttexttexttexttexttexttexttexttext
 the end
+really
 
End Diff
ALERT END


 [test_dir/lnk3] : ALERT: New object
Path:            test_dir/lnk3
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 9
Mode:            lrwxrwxrwx
Target:          test5.txt
Valid:           true
Type:            symbolic link
ALERT END


 [test_dir/test5.txt] : ALERT: New object
Path:            test_dir/test5.txt
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 6
Mode:            -rw-r--r--
SHA256 checksum: 87790b12003ec58e0a786fc4aaeeb11c6d3bbbfa19f0d05fa2368771a7b83ffb
Type:            regular file
ALERT END


 [test_dir/testsub] : ALERT: New object
Path:            test_dir/testsub
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 4096
Mode:            drwxr-xr-x
Type:            directory
ALERT END


 [test_dir/testsub/testsubsub] : ALERT: New object
Path:            test_dir/testsub/testsubsub
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 4096
Mode:            drwxr-xr-x
Type:            directory
ALERT END


 [test_dir/testsub/txt.txt] : ALERT: New object
Path:            test_dir/testsub/txt.txt
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 5
Mode:            -rw-r--r--
SHA256 checksum: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2
Type:            regular file
ALERT END


 [test_dir/testsub/txt3.txt] : ALERT: New object
Path:            test_dir/testsub/txt3.txt
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 676
Mode:            -rw-r--r--
SHA256 checksum: c19f94967b87bf330cc058a16a6eb18ce0bc761c5e2611bf213fc04db79a9f23
Type:            regular file
ALERT END


 [test_dir/txt2.txt] : ALERT: New object
Path:            test_dir/txt2.txt
User:            ITEST_USER_REPLACEME
Group:           ITEST_USER_REPLACEME
Size (in bytes): 5
Mode:            -rw-r--r--
SHA256 checksum: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2
Type:            regular file
ALERT END


 [test_dir/lnk1] : ALERT: object deleted
Path:            test_dir/lnk1
Type:            symbolic link
ALERT END


 [test_dir/test_rm] : ALERT: object deleted
Path:            test_dir/test_rm
Type:            regular file
ALERT END


 [test_dir/txt3.txt] : ALERT: object deleted
Path:            test_dir/txt3.txt
Type:            regular file
ALERT END


